Add alias support to Security Analytics (#6284)
opensearch-project/documentation-website: _security-analytics/sec-analytics-config/detectors-config.md

1. On the **Security Analytics** home page or the **Detectors** page, choose **Create detector**.
1. Give the detector a name and, optionally, a description. 

> 1. In the **Data source** section, select one or more sources for the log data. Use an asterisk (*) to indicate a wildcard pattern. When selecting multiple data sources, their logs must be of the same type. We recommend creating separate detectors for different log types. Security Analytics also supports the following:

Reviewer: ```suggestion
1. In the **Data source** section, select one or more sources for the log data. Use an asterisk (*) to indicate a wildcard pattern. When selecting multiple data sources, their logs must be of the same type. It is good practice to create separate detectors for different log types. Security Analytics also supports the following:
```

>   - [Aliases]({{site.url}}{{site.baseurl}}/im-plugin/index-alias/). When configuring an alias as data source, it must be attached to a **Write** index alias. Furthermore, when using an alias, ensure that your documents are ingesting through the alias and **not** the index(es) the alias was created for.

Reviewer: ```suggestion
  - [Aliases]({{site.url}}{{site.baseurl}}/im-plugin/index-alias/). When configuring an alias as the data source, it must be attached to a **Write** index alias. When using an alias, ensure that your documents are ingesting through the alias and **not** through the index for which the the alias was created.
```

>   - [Data streams]({{site.url}}{{site.baseurl}}/im-plugin/data-streams/) for time-series generated data.

Reviewer 2: @Naarcha-AWS I removed one word, but otherwise LGTM!

1. On the **Security Analytics** home page or the **Detectors** page, choose **Create detector**.
1. Give the detector a name and, optionally, a description. 
1. In the **Data source** section, select one or more sources for the log data. Use an asterisk (*) to indicate a wildcard pattern. When selecting multiple data sources, their logs must be of the same type. We recommend creating separate detectors for different log types. 

>    Data sources support the use of [aliases]({{site.url}}{{site.baseurl}}/im-plugin/index-alias/). However, if you're configuring an alias as data source, it must be attached to a **Write** index alias, since Security Analytics only supports [data stream]({{site.url}}{{site.baseurl}}/im-plugin/data-streams/) for time-series generated data. Furthermore, when using an alias, ensure that your documents are ingesting through the alias and **not** the index(es) the alias was created for. 

Reviewer 3: This part `since Security Analytics only supports [data stream]` might be a little misleading? We support both aliases and data streams independently. A data stream by definition will have a `write` index. However, aliases in opensearch are also allowed to point to read only indices which is why we want to ensure that if a detector is configured on an alias, then it needs point to at least one write index.

>    Data sources support the use of [aliases]({{site.url}}{{site.baseurl}}/im-plugin/index-alias/). However, if you're configuring an alias as data source, it must be attached to a **Write** index alias, since Security Analytics only supports [data stream]({{site.url}}{{site.baseurl}}/im-plugin/data-streams/) for time-series generated data. Furthermore, when using an alias, ensure that your documents are ingesting through the alias and **not** the index(es) the alias was created for. 

Author: @jowg-amazon: Is this more accurate?

```suggestion
   Data sources support the use of [aliases]({{site.url}}{{site.baseurl}}/im-plugin/index-alias/). However, if you're configuring an alias as data source, it must be attached to a **Write** index alias, since all [data streams]({{site.url}}{{site.baseurl}}/im-plugin/data-streams/) require a write index for time-series generated data. Furthermore, when using an alias, ensure that your documents are ingesting through the alias and **not** the index(es) the alias was created for. 
```

Reviewer 3: I think `since all [data streams]({{site.url}}{{site.baseurl}}/im-plugin/data-streams/) require a write index ` part might still be a little misleading? Aliases and data streams do not depend on each other. Aliases require a write index because security analytics detectors are optimized so they only detect on write indices. Data streams happen to not need this additional note because they have a write index configured by definition. Maybe we can keep these two concepts separate and just simply mention that security analytics also supports data streams as a data source?

Author: Let me know if this latest update works.

Reviewer 3: LGTM!
